crypto/evp/e_aes_cbc_hmac_sha256.c

```diff
@@ -453,10 +453,13 @@ static int aesni_cbc_hmac_sha256_cipher(EVP_CIPHER_CTX *ctx,
          * to identify it and avoid stitch invocation. So that after we
          * establish that current CPU supports AVX, we even see if it's
          * either even XOP-capable Bulldozer-based or GenuineIntel one.
+         * But SHAEXT-capable go ahead...
          */
-        if (OPENSSL_ia32cap_P[1] & (1 << (60 - 32)) && /* AVX? */
-            ((OPENSSL_ia32cap_P[1] & (1 << (43 - 32))) /* XOP? */
-             | (OPENSSL_ia32cap_P[0] & (1<<30))) &&    /* "Intel CPU"? */
+        if (((OPENSSL_ia32cap_P[2] & (1<<29)) ||           /* SHAEXT? */
+             ((OPENSSL_ia32cap_P[1] & (1 << (60 - 32))) && /* AVX? */
+              ((OPENSSL_ia32cap_P[1] & (1 << (43 - 32)))   /* XOP? */
+               | (OPENSSL_ia32cap_P[2] & (1<<29))          /* SHAEXT? */
+               | (OPENSSL_ia32cap_P[0] & (1<<30))))) &&    /* "Intel CPU"? */
```

Comment:
It can be either, but | is conscious choice. Rationale is that it will [guarantee to] reduce amount of conditional branches. Less potentially mispredicted branches better.